dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/programrule/DefaultProgramRuleService.java

```diff
@@ -202,10 +212,22 @@ private List<TrackedEntityAttributeValue> getAttributes(
   // if they are present in both places
   private Set<Event> getEventsFromEnrollment(
       String enrollmentUid, TrackerBundle bundle, TrackerPreheat preheat) {
-    Stream<Event> events =
-        preheat.getEvents().values().stream()
-            .filter(e -> e.getEnrollment().getUid().equals(enrollmentUid))
-            .filter(e -> bundle.findEventByUid(e.getUid()).isEmpty());
+    Stream<Event> events;
+    try {
+      events =
+          eventService
+              .getEvents(
+                  EventOperationParams.builder()
+                      .eventParams(EventParams.TRUE)
+                      .orgUnitMode(ACCESSIBLE)
```

Comment:
Who is the currentUser/who set it when this is executed?

Reply:
This is using current user.
Exporters uses current user to perform validations.

Reply:
but the program rules run as part of the import job which sets the current user to the user making the HTTP request, right?

Reply:
Yes yes, program rules run as part of import job, so they are using the same user.